### PR TITLE
Fixes for the stats endpoint

### PR DESCRIPTION
**Pull request by arielsvn**
## Issue Number

https://github.com/AlexsLemonade/refinebio-frontend/issues/566

## Purpose/Implementation Notes

Add improvements to stats endpoint that were suggested on https://github.com/AlexsLemonade/refinebio-frontend/issues/566. The most important ones are:

- Use `last_modified` when calculating stats on processed samples
- Fix nomad jobs aggregation by volume.

## Types of changes

- Bugfix (non-breaking change which fixes an issue)

## Functional tests

Not sure why tests are failing ... seems the mocks don't work. WIll look into it

## Checklist

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by arielsvn** [comment]
@Miserlou @kurtwheeler I'm trying to mock a function for the tests here, but I can't get it to work. Is there something wrong with this ?

https://github.com/AlexsLemonade/refinebio/blob/b607fc0707b97008319699039868ea6c6bdebd20/api/data_refinery_api/tests.py#L1148-L1151

**Comment by kurtwheeler** [comment]
@arielsvn I can help you with that. It's because you're mocking where it's coming from, not where it's being used.

File: api/data_refinery_api/views.py [hunk 1]
```diff
@@ -1174,39 +1172,28 @@ def _get_job_details(self, job):
...
-        try:
-            nomad_host = get_env_variable("NOMAD_HOST")
-            nomad_port = get_env_variable("NOMAD_PORT", "4646")
-            nomad_client = nomad.Nomad(nomad_host, port=int(nomad_port), timeout=30)
-            return nomad_client.jobs.get_jobs()
-
-        except nomad.api.exceptions.BaseNomadException:
-            # Nomad is not available right now
-            return []
+    def _get_job_type(self, job): return self._get_job_details(job)[0]
```
Comment (Miserlou) [inline api/data_refinery_api/views.py]: I don't like these

Comment (arielsvn) [reply api/data_refinery_api/views.py]: Do you think it's better to have it as a variable with a lambda function?

```
get_job_type = lambda job: self._get_job_details(job)[0]
```

**Comment by arielsvn** [comment]
@Miserlou changed this to use lambda functions. Let me know what do you think.

**Pull request merged**